### PR TITLE
Document relationships and querying capabilites

### PR DESCRIPTION
**Pull request by kriszyp**

### CODE REVIEW

File: docs/developers/applications/defining-schemas.md [hunk 1]
```diff
@@ -53,6 +53,34 @@ type MyTable @table @export(name: "my-table")
...
+#### `@relationship(from: attribute)`
+This defines a relationship where the foreign key is defined in this table, and relates to the primary key of the target table. If the foreign key is single-valued, this establishes a many-to-one relationship with the target table. The foreign key may also be a multi-valued array, in which case this will be a many-to-many relationship.
+For example, we can define a foreign key that references another table and then define the relationship. Here we create a `brandId` attribute that will be our foreign key (it will hold an id that references the primary key of the Brand table), and we define a relationship to the `Brand` table through the `brand` attribute:
+```graphql
+type Product @table @export {
+	id: ID @primaryKey
+	brandId: ID @indexed
+	brand: Brand @relationship(from: brandId)
+}
+type Brand @table @export {
```
Comment (jcohen-hdb) [inline docs/developers/applications/defining-schemas.md:68]: Should we include the Brand table primary key as well? I feel like that completes the picture.

File: docs/developers/operations-api/nosql-operations.md [hunk 2]
```diff
@@ -273,20 +273,26 @@ Returns data from a table for a matching value.
...
-Returns data from a table for one or more matching conditions.
+Returns data from a table for one or more matching conditions. This supports grouping of conditions to indicate order of operations as well.
 
 * operation _(required)_ - must always be `search_by_conditions`
 * database _(optional)_ - database where the table you are searching lives. The default is `data`
 * table _(required)_ - table you wish to search
 * operator _(optional)_ - the operator used between each condition - `and`, `or`. The default is `and`
 * offset _(optional)_ - the number of records that the query results will skip. The default is `0`
 * limit _(optional)_ - the number of records that the query results will include. The default is `null`, resulting in no limit
+* sort _optional_ - This is an object that indicates the sort order. It has the following properties:
```
Comment (jcohen-hdb) [inline docs/developers/operations-api/nosql-operations.md:284]: Wooooo, sorting!!!

File: docs/developers/rest.md [hunk 3]
```diff
@@ -116,35 +116,119 @@ GET /my-resource/?age=le=20
...
+```http
+GET /Product?rating=gt=3&sort(+name)
+```
+To sort by rating in ascending order, then by price in descending order for products with the same rating:
+```http
+GET /Product?sort(+rating,-price)
+```
+
+## Nested Attributes and Joins
+In addition to querying on top-level attributes, you can also query on nested attributes. Most importantly, this provides "join" functionality, allowing related tables to be queried and joined in the results. Nested properties are specified by using dot syntax. In order to effectively leverage join functionality, you need to define a relationship in your schema:
```
Comment (jcohen-hdb) [inline docs/developers/rest.md:192]: I read nested attributes as:
```
{
  "name": "Harper",
  "traits": {
    "goofy": true,
    "smart": true,
    "crazy" true
  }
}
```
Meaning it's nested JSON. So for this, I might call it joined attributes or something like that? I'm not exactly sure, just giving you my immediate impression. It still makes sense once I read it all the way through.

**Review by jcohen-hdb — COMMENTED** [review]
Added a few comments. Ultimately, I think it's good as is, but wanted to run these by you before approving. Let me know!

File: docs/developers/rest.md (hunk 3 above)
Comment (kriszyp) [reply docs/developers/rest.md]: Maybe "chained attributes"?

**Pull request merged**